### DISCUSSION TITLE
Generation of stairs with IfcOpenShell python

### DISCUSSION THREAD

**Discussion by NkStll** [IfcOpenShell/IfcOpenShell#3116] [Q&A]
Dear all,
I am trying to generate a flight of stairs with ifcopenshell in python. Is there some kind of autmation for this or do I need to generate it step by step and add landings and railings manually?
Any help is greatly appreciated.
Thank you!

**Comment by Moult** [comment]
There is code for this in the BlenderBIM Add-on but it has not yet been generalised to be a pure Python implementation.
